### DISCUSSION TITLE
Digital signature verification

### DISCUSSION THREAD

**Discussion by HeshimaPrime** [akopachov/flipper-zero_authenticator#159] [Q&A] [answered]
Hey, when ever I try to launch the app I get a screen that says "Digital signature verification - failed". How do I go about fixing that?

**Comment by akopachov** [comment]
What is the firmware and version you are running? What is the version of Flipper Authenticator?

**Reply by Baatezu** [reply]
same problem firmware: Version: unlshd-055, not sure about Flipper Authenticator but in change log say they use last version

**Reply by akopachov** [reply]
Let's see if latest version [v2.2.2](https://github.com/akopachov/flipper-zero_authenticator/releases/tag/v2.2.2) going to work for you.

**Reply by Baatezu** [reply]
"reinstall" last version same problem

**Reply by akopachov** [reply]
Have you changed Flipper name? if so, when? (before\after you initially run Authenticator)

**Reply by HeshimaPrime** [reply]
I was also using unlshd-055 and I tried the running the version that came installed with it as well as reinstalling the latest version of the authenticator. I just now tried running it with XFW-0048 and still have the same problem. I have not changed the name of my flipper and I got this error the first time I tried running the authenticator.

**Reply by akopachov** [reply, marked as answer]
if you ran into this the first time you are trying to use this app and you are definitely not used it before and didn't add any tokens there, then just drop `/ext/authenticator/totp.conf` file. This will let you start from scratch.

**Reply by HeshimaPrime** [reply]
I tried it and it works now, thank you.

**Reply by Baatezu** [reply]
Not working for me, if I delete .conf its loading but its not working with my old .conf

**Reply by Baatezu** [reply]
If I install v2.0.3 it's working. try to guess problem maybe because before pin is not necessary and in new version its should be?

**Reply by akopachov** [reply]
Not really. Ability to setup PIN was in this app since first alpha-versions. And since than, PIN was optional. It is up to a user whether to setup it or not. BUT just in case if user decided to do not setup PIN, then app will use device UID. To get that device UID, before version 2.2.0 firmware function `furi_hal_version_uid` was used, but then it was discovered [here](https://github.com/akopachov/flipper-zero_authenticator/issues/153) that some firmwares updated this function to return "fake" uid in certain conditions. So in version 2.2.0 I updated the code to do not use `furi_hal_version_uid` function and instead read it directly.
However, not all the firmwares merged this change to Authenticator they are bundling as a preinstalled software.

So there is a chance that your initial totp.conf file was created using "faked" uid, and now you are trying to access it with version of the app which is using real uid.

The best what I can offer in such a situation is to set PIN using version 2.0.3 by using `totp pin set` CLI command.
Then install latest version and use `totp pin remove` to remove PIN and let app use your device real uid.

This trick should re-encrypt all the things using new UID and after that you should be able to use new version normally.

Let me know if any issues.
